lookup_test.go

```diff
@@ -50,7 +52,7 @@ func TestSimpleLookup(t *testing.T) {
 				ValidPtr *string
 				NilPtr   *string
 			}{
-				123, "abc", true, struct{ Inside string }{"I'm nested!"}, "xyz", "bad", "bad", &stringForPtr, nil,
+				123, "abc", [3]int{1, 2, 3}, []int{1}, true, struct{ Inside string }{"I'm nested!"}, "xyz", "bad", "bad", &stringForPtr, nil,
```

Comment:
Can you add a test for nested array lookup? For example:
array: []struct{
  subArray []int   
}{
  {
    []int{1},
  },
{
    []int{2},
  },
}

and then Array.0.subArray.0

I think it should just work, but seems worth to have a test for this case